pkg/xds/generator/admin_proxy_generator.go

```diff
@@ -41,13 +63,39 @@ func (g AdminProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds.P
 
 	resources := core_xds.NewResourceSet()
 
+	for _, se := range staticEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+	}
+
+	for _, se := range staticTlsEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+
+		token, err := ctx.EnvoyAdminClient.GenerateAPIToken(proxy.Dataplane)
+		if err != nil {
+			return nil, err
+		}
+		se.Header = "Authorization"
+		se.HeaderExactMatch = fmt.Sprintf("Bearer %s", token)
+	}
+
+	keyPair, err := NewSelfSignedCert("admin", tls.ServerCertType, "localhost")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I don't think it is a good idea to generate self-signed cert for every dataplane on every xds generation loop.
Please generate it once when we build ControlPlaneContext and put it there.